woocommerce/woocommerce
# Marketplace: modernize skeleton loaders

### Changes proposed in this Pull Request:

Closes [WCCOM#18034](https://github.com/Automattic/woocommerce.com/issues/18034). Closes [WCCOM#18257](https://github.com/Automattic/woocommerce.com/issues/18257).

This PR replaces all "skeleton" loading indicators on the Marketplace with more-standard ones. It also aims to make those skeleton loaders:
- More-accurately represent this size and shape of the content that will replace them, and
- Refactors the code so that components are responsible for hosting their own skeleton code, attached to an `isLoading` variable, making it easier to stay consistent as changes are made in future

### How to test the changes in this Pull Request:

1. Check out this branch and ensure everything's recompiled with e.g. `cd plugins/woocommerce-admin && pnpm run start:hot`
2. Visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions and observe the new skeleton loaders on the Discover page
3.Visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions and see the new skeleton loaders on the Browse page (you might also like to do a search)

### Demo video

This demo video artificially slows down loading a _lot_ so the skeletons are very visible:

https://github.com/woocommerce/woocommerce/assets/53293/0b17682b-a87e-4b81-bcaf-abe835be9c5f

### Changelog entry


<details>

#### Significance


#### Type


#### Message 

#### Comment 

Improved loading indicators on WooCommerce > Extensions so they more-accurately reflect the layout of the content that will replace them.

</details>
